nsxt/resource_nsxt_policy_ods_runbook_invocation.go

```diff
@@ -0,0 +1,202 @@
+	description := d.Get("description").(string)
+	tags := getPolicyTagsFromSchema(d)
+	runbookPath := d.Get("runbook_path").(string)
+	targetNode := d.Get("target_node").(string)
+
+	var arguments []model.UnboundedKeyValuePair
+	for _, arg := range d.Get("argument").(*schema.Set).List() {
+		argMap := arg.(map[string]interface{})
+		key := argMap["key"].(string)
+		value := argMap["value"].(string)
+		item := model.UnboundedKeyValuePair{
+			Key:   &key,
+			Value: &value,
+		}
+		arguments = append(arguments, item)
+	}
+
+	obj := model.OdsRunbookInvocation{
+		Id:          &id,
+		Description: &description,
+		Tags:        tags,
+		RunbookPath: &runbookPath,
+		Arguments:   arguments,
+		TargetNode:  &targetNode,
+	}
+	if displayName != "" {
+		obj.DisplayName = &displayName
+	}
+
+	return obj
+}
+
+func resourceNsxtPolicyODSRunbookInvocationCreate(d *schema.ResourceData, m interface{}) error {
+	// Initialize resource Id and verify this ID is not yet used
+	id, err := getOrGenerateID(d, m, resourceNsxtPolicyODSRunbookInvocationExists)
+	if err != nil {
+		return err
+	}
+
+	connector := getPolicyConnector(m)
+	client := sha.NewRunbookInvocationsClient(connector)
+
+	obj := getODSRunbookInvocationFromSchema(id, d)
+	err = client.Create(id, obj)
+	if err != nil {
+		return handleCreateError("OdsRunbookInvocation", id, err)
+	}
+
+	d.SetId(id)
+	d.Set("nsx_id", id)
+	return resourceNsxtPolicyODSRunbookInvocationRead(d, m)
+}
+
+func resourceNsxtPolicyODSRunbookInvocationExists(id string, connector client.Connector, isGlobalManager bool) (bool, error) {
+	var err error
+	client := sha.NewRunbookInvocationsClient(connector)
+	_, err = client.Get(id)
+
+	if err == nil {
+		return true, nil
+	}
+
+	if isNotFoundError(err) {
+		return false, nil
+	}
+
+	return false, logAPIError("Error retrieving resource", err)
+}
+
+func resourceNsxtPolicyODSRunbookInvocationRead(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("error obtaining OdsRunbookInvocation ID")
+	}
+
+	client := sha.NewRunbookInvocationsClient(connector)
+	var err error
+	obj, err := client.Get(id)
+	if err != nil {
+		return handleReadError(d, "OdsRunbookInvocation", id, err)
+	}
+
+	if obj.DisplayName != nil && *obj.DisplayName != "" {
+		d.Set("display_name", obj.DisplayName)
+	}
+	d.Set("description", obj.Description)
+	setPolicyTagsInSchema(d, obj.Tags)
+	d.Set("nsx_id", id)
+	d.Set("path", obj.Path)
+	d.Set("revision", obj.Revision)
+
+	d.Set("runbook_path", obj.RunbookPath)
+	d.Set("target_node", obj.TargetNode)
+
+	var argList []map[string]interface{}
+	for _, arg := range obj.Arguments {
+		argData := make(map[string]interface{})
+		argData["key"] = arg.Key
+		argData["value"] = arg.Value
+		argList = append(argList, argData)
+	}
+	d.Set("argument", argList)
+
+	return nil
+}
+
+func resourceNsxtPolicyODSRunbookInvocationUpdate(d *schema.ResourceData, m interface{}) error {
```

Comment:
If the user changes description, this will result in constant non-empty diff. Do tags/description have any value for this resource, perhaps we can remove them?

Reply:
Makes sense, if anyone ever needs them we can reevaluate how to handle these.